packages/shopify-checkout/src/utils/metafieldsToCustomAttributes.ts

```diff
@@ -17,7 +17,9 @@ export default function metafieldsToCustomAttributes({
   }
 
   const customAttributes: Attribute[] = metafields?.reduce((fields, m) => {
-    fields.push({ key: m.key, value: m.value });
+    if (typeof m.value === 'string') {
+      fields.push({ key: m.key, value: m.value });
+    }
```

Comment:
One possibility here would be to stringify non-string values. For example:

```ts
if (typeof m.value === 'string') {
  fields.push({ key: m.key, value: m.value });
} else {
  fields.push({ key: m.key, value: JSON.stringify(m.value) });
}
```

It looks like that would work equally well for numbers, arrays, and objects:

<img width="400" alt="JSON stringify number, array, and object" src="https://user-images.githubusercontent.com/5732000/146420287-52c028d2-b1ad-4703-99ce-37e0d670e794.png">

Reply:
@NWRichmond I considered that.  My thought was merchants might not want to pass along "undefined".  I guess I'm not too opinionated here.

Reply:
For sure. Makes me wonder if we should instead:

```ts
if (typeof m.value === 'string') {
  fields.push({ key: m.key, value: m.value });
} else {
  console.warn(`Omitting custom attribute "${key}" because it has a non-string value.`);
}
```

Reply:
Sounds good to me!  I'll get that into this PR and tested

Reply:
@NWRichmond since the `metafield.value` is typed as a string, I'm not able to test this easily.  Basically the test fails with Typescript errors.